test/NFTDescriptor.spec.ts

```diff
@@ -389,6 +396,46 @@ describe('NFTDescriptor', () => {
           expect(await nftDescriptor.fixedPointToDecimalString(ratio, 24, 5)).to.eq('1.0000')
         })
       })
+
+      it('some fuzz', async () => {
+        const random3To21 = () => {
+          return Math.floor(3 + ((Math.random() * 100) % 18))
+        }
+        const random1To20 = () => {
```

Comment:
looks like you probably just want something like `rand(min: number, maxExclusive: number)`